eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -539,25 +537,66 @@ object Helpers {
       }.flatten
 
       // all htlc output to us are delayed, so we need to claim them as soon as the delay is over
-      val htlcDelayedTxes = htlcTxes.flatMap {
-        txinfo: TransactionWithInputInfo =>
-          generateTx("claim-htlc-delayed") {
-            Transactions.makeClaimLocalDelayedOutputTx(txinfo.tx, localParams.dustLimit, localRevocationPubkey, remoteParams.toSelfDelay, localDelayedPubkey, localParams.defaultFinalScriptPubKey, feeratePerKwDelayed).right.map(claimDelayed => {
-              val sig = keyManager.sign(claimDelayed, keyManager.delayedPaymentPoint(channelKeyPath), localPerCommitmentPoint, TxOwner.Local, commitmentFormat)
-              Transactions.addSigs(claimDelayed, sig)
-            })
+      // NB: when using anchor outputs, we will claim them once the corresponding HTLC txs confirm
+      val htlcDelayedTxs = commitmentFormat match {
+        case Transactions.DefaultCommitmentFormat =>
+          htlcTxs.flatMap {
+            txinfo: TransactionWithInputInfo =>
+              generateTx("claim-htlc-delayed") {
+                Transactions.makeClaimLocalDelayedOutputTx(txinfo.tx, localParams.dustLimit, localRevocationPubkey, remoteParams.toSelfDelay, localDelayedPubkey, localParams.defaultFinalScriptPubKey, feeratePerKwDelayed).map(claimDelayed => {
+                  val sig = keyManager.sign(claimDelayed, keyManager.delayedPaymentPoint(channelKeyPath), localPerCommitmentPoint, TxOwner.Local, commitmentFormat)
+                  Transactions.addSigs(claimDelayed, sig)
+                })
+              }
           }
+        case Transactions.AnchorOutputsCommitmentFormat => Nil
       }
 
       LocalCommitPublished(
         commitTx = tx,
         claimMainDelayedOutputTx = mainDelayedTx.map(_.tx),
-        htlcSuccessTxs = htlcTxes.collect { case c: HtlcSuccessTx => c.tx },
-        htlcTimeoutTxs = htlcTxes.collect { case c: HtlcTimeoutTx => c.tx },
-        claimHtlcDelayedTxs = htlcDelayedTxes.map(_.tx),
+        htlcSuccessTxs = htlcTxs.collect { case c: HtlcSuccessTx => c.tx },
+        htlcTimeoutTxs = htlcTxs.collect { case c: HtlcTimeoutTx => c.tx },
+        claimHtlcDelayedTxs = htlcDelayedTxs.map(_.tx),
         irrevocablySpent = Map.empty)
     }
 
+    /**
+     * Claim the output of a 2nd-stage HTLC transaction and replace the obsolete HTLC transaction in our local commit.
+     */
```

Comment:
```suggestion
    /**
     * Claim the output of a 2nd-stage HTLC transaction and replace the obsolete HTLC transaction in our local commit
     * Only used in the context of the anchor output commit format. If the provided transaction isn't an htlc, this will be a no-op.
     */
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1702/commits/5d88187901aa9a1892726540ff68d82dc77a7961